app/src/main/java/fr/free/nrw/commons/coordinates/CoordinateEditHelper.java

```diff
@@ -0,0 +1,178 @@
+import fr.free.nrw.commons.notification.NotificationHelper;
+import fr.free.nrw.commons.utils.ViewUtilWrapper;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import io.reactivex.schedulers.Schedulers;
+import java.util.Objects;
+import javax.inject.Inject;
+import javax.inject.Named;
+import org.apache.commons.lang3.StringUtils;
+import timber.log.Timber;
+
+/**
+ * Helper class for edit and update given coordinates and showing notification about new coordinates
+ * upgradation
+ */
+public class CoordinateEditHelper {
+
+    /**
+     * notificationHelper: helps creating notification.
+     * pageEditClient: methods provided by this member posts the edited coordinates
+     * to the Media wiki api.
+     * viewUtil: helps to show Toast
+     */
+    private final NotificationHelper notificationHelper;
+    public final PageEditClient pageEditClient;
+    private final ViewUtilWrapper viewUtil;
+
+    @Inject
+    public CoordinateEditHelper(final NotificationHelper notificationHelper,
+        @Named("commons-page-edit") final PageEditClient pageEditClient,
+        final ViewUtilWrapper viewUtil) {
+        this.notificationHelper = notificationHelper;
+        this.pageEditClient = pageEditClient;
+        this.viewUtil = viewUtil;
+    }
+
+    /**
+     * Public interface to edit coordinates
+     * @param context to be added
+     * @param media to be added
+     * @param Accuracy to be added
+     * @return Single<Boolean>
+     */
+    public Single<Boolean> makeCoordinatesEdit(final Context context, final Media media,
+        final String Latitude, final String Longitude, final String Accuracy) {
+        viewUtil.showShortToast(context,
+            context.getString(R.string.coordinates_edit_helper_make_edit_toast));
+        return addCoordinates(media, Latitude, Longitude, Accuracy)
+            .flatMapSingle(result -> Single.just(showCoordinatesEditNotification(context, media,
+                Latitude, Longitude, Accuracy, result)))
+            .firstOrError();
+    }
+
+    /**
+     * Replaces new coordinates
+     * @param media to be added
+     * @param Latitude to be added
+     * @param Longitude to be added
+     * @param Accuracy to be added
+     * @return Observable<Boolean>
+     */
+    private Observable<Boolean> addCoordinates(final Media media, final String Latitude,
+        final String Longitude, final String Accuracy) {
+        Timber.d("thread is coordinates adding %s", Thread.currentThread().getName());
+        final String summary = "Adding Coordinates";
+
+        final StringBuilder buffer = new StringBuilder();
+
+        final String wikiText = pageEditClient.getCurrentWikiText(media.getFilename())
+            .subscribeOn(Schedulers.io())
+            .blockingGet();
+
+        if (Latitude != null) {
+            buffer.append("\n{{Location|").append(Latitude).append("|").append(Longitude)
+                .append("|").append(Accuracy).append("}}\n");
```

Comment:
Are the two `\n` really necessary? Better remove them as they make the difference bigger than necessary: https://commons.wikimedia.org/w/index.php?title=File:Hamamatsucho_station_pedestrian_bridge_seen_from_moving_monorail.jpg&diff=prev&oldid=564285824